Frends.FTP.DownloadFiles/Frends.FTP.DownloadFiles/Logging/FtpLogger.cs

```diff
@@ -5,182 +5,182 @@
 
-            return new FileTransferInfo
-            {
-                Result = result,
-                SourceFile = sourceFile ?? string.Empty,
-                DestinationFile = destinationFile ?? string.Empty,
-                FileSize = GetFileSize(localFileName),
-                TransferStarted = transferStarted,
-                TransferEnded = DateTime.UtcNow,
-                BatchId = batchId,
-                TransferName = transferName ?? string.Empty,
-                ErrorInfo = errorMessage ?? string.Empty
-            };
-        }
+    public static FileTransferInfo CreateFileTransferInfo(TransferResult result, SingleFileTransfer transfer, BatchContext context, string errorMessage = null)
+    {
+        // Create 2 dummy endpoints and initialize some local variables which are needed in case if cobalt config is not
+        // successfully initialized, i.e. when there has been a failure creating the config (invalid xml etc..) and config elements are left null
+        var sourceFile = string.Empty;
+        var destinationFile = string.Empty;
+        var localFileName = string.Empty;
 
-        private static long GetFileSize(string filepath)
+        if (transfer != null)
         {
-            return File.Exists(filepath) ? new FileInfo(filepath).Length : 0;
+            sourceFile = transfer.SourceFile.Name;
+            destinationFile = transfer.DestinationFileNameWithMacrosExpanded;
+            localFileName = context.Info.WorkDir;
         }
 
-        public void Dispose()
+        var transferStarted = DateTime.UtcNow;
+        var batchId = Guid.Empty;
+
+        var transferName = string.Empty;
+
+        if (context != null)
         {
-            Dispose(true);
-            GC.SuppressFinalize(this);
+            transferStarted = context.BatchTransferStartTime;
+            batchId = context.InstanceId;
+            transferName = context.Info != null ? context.Info.TransferName : string.Empty;
         }
 
-        private void Dispose(bool disposing)
+        return new FileTransferInfo
         {
-            if (_disposed)
-            {
-                return;
-            }
+            Result = result,
+            SourceFile = sourceFile ?? string.Empty,
+            DestinationFile = destinationFile ?? string.Empty,
+            FileSize = GetFileSize(localFileName),
+            TransferStarted = transferStarted,
+            TransferEnded = DateTime.UtcNow,
+            BatchId = batchId,
+            TransferName = transferName ?? string.Empty,
+            ErrorInfo = errorMessage ?? string.Empty
+        };
+    }
 
-            _fileTransfers = null;
-            _log = null;
+    private static long GetFileSize(string filepath)
+    {
+        return File.Exists(filepath) ? new FileInfo(filepath).Length : 0;
+    }
 
-            _disposed = true;
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    private void Dispose(bool disposing)
```

Comment:
You're not using parameter `disposing`, so it can be removed.

Reply:
This needs to be implemented in order to use the interface IFTPLogger with is using IDisposable. I'm not sure if this is needed.